sdk/aqueduct/aqueduct_client.py

```diff
@@ -449,22 +449,19 @@ def delete_flow(
             flow_id, saved_objects_to_delete, force
         )
 
-        failed_deletions = {}
-        counts = 0
+        failures = []
         for integration in resp.saved_object_deletion_results:
-            failed_for_integration = []
             for obj in resp.saved_object_deletion_results[integration]:
-                if not obj.succeeded:
-                    failed_for_integration.append(obj)
-            if len(failed_for_integration) > 0:
-                failed_deletions[str(integration)] = [
-                    res.__dict__ for res in failed_for_integration
-                ]
-                counts += len(failed_for_integration)
-        if counts > 0:
-            failures = json.dumps(failed_deletions, sort_keys=False, indent=4)
+                if obj.exec_state.status == ExecutionStatus.FAILED:
+                    assert isinstance(obj.exec_state.error, Error)
```

Comment:
why do we need an `assert` here? I remembered the field is just `Optional[Error]`? If the type mismatch the serialization might just doesn't work. If it's only for detecting if the field is not `None`, we should just use `if  not obj.exec_state.error`

Reply:
Yeah that makes sense, thanks!